src/csharp/Microsoft.Spark/Utils/PythonSerDe.cs

```diff
@@ -67,5 +67,31 @@ internal static object[] GetUnpickledObjects(Stream stream, int messageLength)
                 ArrayPool<byte>.Shared.Return(buffer);
             }
         }
+
+        /// <summary>
+        /// Custom pickler for GenericRow objects.
+        /// </summary>
+        internal class GenericRowPickler : IObjectPickler
+        {
+            public void pickle(object o, Stream outs, Pickler currentPickler)
+            {
+                GenericRow row = (GenericRow)o;
+                currentPickler.save(row.Values);
+            }
+
+            public void Register()
+            {
+                Pickler.registerCustomPickler(typeof(GenericRow), new GenericRowPickler());
+            }
+        }
+
+        /// <summary>
+        /// Register custom pickler for GenericRow object and create pickler.
+        /// </summary>
+        internal static Pickler CreatePickler()
+        {
+            new GenericRowPickler().Register();
+            return new Pickler();
+        }
```

Comment:
Why are these new methods necessary ?  Can't the `GenericRowPickler` be registered in the same location where the `RowConstructor` is registered ?

Reply:
Updated. Thanks!